src/main/scala/monix/nio/tcp/AsyncTcpClientConsumer.scala

```diff
@@ -0,0 +1,128 @@
+package monix.nio.tcp
+
+import java.net.InetSocketAddress
+import java.nio.ByteBuffer
+
+import monix.eval.Callback
+import monix.execution.Ack.{Continue, Stop}
+import monix.execution.{Ack, Scheduler, UncaughtExceptionReporter}
+import monix.execution.atomic.Atomic
+import monix.execution.cancelables.{AssignableCancelable, SingleAssignmentCancelable}
+import monix.nio.cancelables.SingleFunctionCallCancelable
+import monix.reactive.Consumer
+import monix.reactive.observers.Subscriber
+
+import scala.concurrent.{Future, Promise}
+import scala.util.control.NonFatal
+
+class AsyncTcpClientConsumer private[tcp] (host: String, port: Int) extends Consumer[Array[Byte], Long] {
+
+  private[this] var socketClient: Option[SocketClient] = None
+
+  private[tcp] def this(client: SocketClient) {
+    this(client.to.getHostString, client.to.getPort)
+    this.socketClient = Option(client)
+  }
+
+  override def createSubscriber(cb: Callback[Long], s: Scheduler): (Subscriber[Array[Byte]], AssignableCancelable) = {
```

Comment:
The logic is quite different. For TCP we have to know when the channel is connected and also not close it if one is waiting for a response to. So, not much to abstract over